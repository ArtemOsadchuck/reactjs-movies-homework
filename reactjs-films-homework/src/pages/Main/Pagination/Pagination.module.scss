$wite: rgb(255, 255, 255);
$black: rgb(0, 0, 0);
$gray: rgba(231, 231, 231, 0.775);
$green: rgb(41, 186, 41);

.paginationWrapper {
  bottom: 1rem;
  color: $gray;
  display: flex;
  height: 2rem;
  justify-content: space-between;
  margin-bottom: 1.5rem;
  margin-top: 1rem;
  width: 20rem;

  div {
    align-items: center;
    display: flex;
    height: 2rem;
    justify-content: center;
    transition: 0.1s;
    width: 2rem;

    &:hover {
      background-color: $green;
      border-radius: 50%;
      color: $wite;
      cursor: pointer;
    }
  }
}
