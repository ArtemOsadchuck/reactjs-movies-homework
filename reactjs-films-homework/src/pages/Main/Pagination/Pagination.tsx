import React, { useEffect, useState } from 'react';
import styles from './Pagination.module.scss';
import { useAppSelector, useAppDispatch } from '../../../hooks/hooks';
import getMainData from '../../../store/rootStore/mainStore/getMaiData/getMainData';
import {
  setPage,
  setActivePage,
} from '../../../store/rootStore/mainStore/mainSlice';

const getPagesFromTotal = (totalPages: number) => {
  const resultArr: string[] = [];
  for (let i = 1; i <= totalPages; i += 1) {
    resultArr.push(`${i}`);
  }
  return resultArr.slice(0, 5);
};

const Pagination: React.FC = () => {
  const [pagNum, setPugNum] = useState<Array<string>>([]);
  const dispatch = useAppDispatch();
  const mainState = {
    lang: useAppSelector((state) => state.mainReducer.lang),
    category: useAppSelector((state) => state.mainReducer.category),
    query: useAppSelector((state) => state.mainReducer.query),
    activePage: useAppSelector((state) => state.mainReducer.activePage),
    totalPages: useAppSelector((state) => state.mainReducer.totalPages),
  };

  useEffect(() => {
    setPugNum(() => getPagesFromTotal(mainState.totalPages));
  }, [dispatch, mainState.totalPages]);

  const getPage = (page: string) => {
    dispatch(setActivePage(page));
    dispatch(setPage(page));
    dispatch(getMainData({ ...mainState, page: page }));
    return;
  };

  return (
    <div id="paginationWrapper_testID" className={styles.paginationWrapper}>
      {pagNum.map((page) => {
        return (
          <div
            key={page}
            onClick={() => {
              getPage(page);
            }}
            className={
              `${mainState.activePage}` === `${page}` ? styles.active : ''
            }
          >
            {page}
          </div>
        );
      })}
    </div>
  );
};
export default Pagination;
