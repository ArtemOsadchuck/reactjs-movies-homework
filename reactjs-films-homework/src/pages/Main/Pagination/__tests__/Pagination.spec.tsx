import React from 'react';
import { render, cleanup } from '@testing-library/react';

import Pagination from '../Pagination';

import StoreWrapper from '../../../../__testsUtils__/storeHoc';
import RouterWrapper from '../../../../__testsUtils__/routerHoc';

const renderComponent = () => {
  const urlApp = '/';
  const { asFragment } = render(
    <StoreWrapper>
      <RouterWrapper url={urlApp}>
        <Pagination neededPages={5} />
      </RouterWrapper>
    </StoreWrapper>
  );
  return asFragment();
};

describe('Pagination', () => {
  it('should renders correctly', () => {
    const fragment = renderComponent();

    expect(fragment).toMatchSnapshot();
  });
  cleanup();
});
