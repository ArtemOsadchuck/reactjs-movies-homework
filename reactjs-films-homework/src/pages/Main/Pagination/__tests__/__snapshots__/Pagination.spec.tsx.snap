// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should renders correctly 1`] = `
<DocumentFragment>
  <div
    class="paginationWrapper"
  />
</DocumentFragment>
`;
